File: mwc/romana/relic/d/lib/libc/string/strcpy.c
/*
 * strcpy.c
 * ANSI 4.11.2.3.
 * Copy one string to another.
 */

#include <string.h>

char *strcpy(s1, s2) char *s1; register char *s2;
{
	register char *cp;

	for (cp = s1; *cp++ = *s2++; )
		;
	return (s1);
}

File: mwc/romana/relic/d/lib/libc/string/i8086/strcpy.m
//////////
/ i8086 C string library.
/ strcpy()
/ ANSI 4.11.2.3.
//////////

//////////
/ char *
/ strcpy(To, From)
/ char *To, *From;
/
/ Copy From to To until NUL.
//////////

#include <larges.h>

To	=	LEFTARG
From	=	To+DPL

	Enter(strcpy_)
	Lds	si, From(bp)	/ From address to DS:SI
	Les	di, To(bp)	/ To address to ES:DI
	cld

1:	lodsb			/ From character to AL
	stosb			/ and stored to To
	orb	al, al
	jne	1b		/ Not done
	mov	ax, To(bp)	/ Return the destination
#if	LARGEDATA
	mov	dx, es
#endif
	Leave
